### PR TITLE
Add onboarding squarespace importer

### PR DESCRIPTION
**Pull request by zaerl**
#### Changes proposed in this Pull Request

* Add a Squarespace importer
* Add support for favicon-s on the author mapping screen (added to `ImporterAuthorMapping`)
* Add a `CompleteScreen` and `ProgressScreen` components shared between routes
* Minor refactoring of importer components to pass `UrlData`

#### Testing instructions

* Open http://calypso.localhost:3000/start/importer/capture?siteSlug=SITE_SLUG
* Input a Squarespace URL such as https://squarespace.com/
* Select a Squarespace `.xml` export file; you can find one on _Drive > Caribou > Sample Export Files > Squarespace_
* Import the XML
* Check if the data have been imported

#### Important
The author list component is missing the favicon because Squarespace `analyze-call` returns `null`.

### CODE REVIEW

File: client/signup/steps/import-from/squarespace/index.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+			},
+		};
+
+		importerData.title = translate( 'Import content from %(importerName)s', {
+			...options,
+			textOnly: true,
+		} as TranslateOptionsText ) as string;
+
+		importerData.description = translate(
+			'Import your posts, pages, comments, tags, and images from your %(importerName)s export file.',
```
Comment (a8ci18n) [inline client/signup/steps/import-from/squarespace/index.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **24** times:
`translate( 'Import posts, pages, comments, tags, and images from a %(importerName)s export file.' )` *ES Score: 7*

Comment (zaerl) [reply client/signup/steps/import-from/squarespace/index.tsx]: @vishnugopal @a8ci18n This is only slightly different from the default one. It only has a "your" word added. Maybe we should not have a new one.

Comment (vishnugopal) [reply client/signup/steps/import-from/squarespace/index.tsx]: Agreed, let's replace!

File: client/signup/steps/import-from/components/importer-drag/index.tsx [hunk 2]
```diff
@@ -41,10 +42,11 @@ interface Props {
 	importerStatus: ImportJob;
 	importerData: ImporterConfig;
 	site: Site;
+	urlData: UrlData;
 	startImport: ( siteId: number, type: string ) => void;
 }
 const ImporterDrag: React.FunctionComponent< Props > = ( props ) => {
-	const { importerStatus, importerData, site /*, startImport*/ } = props;
+	const { importerStatus, importerData, site, urlData /*, startImport*/ } = props;
```
Comment (vishnugopal) [inline client/signup/steps/import-from/components/importer-drag/index.tsx:49]: Why is `startImport` commented out?

Comment (vishnugopal) [inline client/signup/steps/import-from/components/importer-drag/index.tsx:49]: Why is `startImport` commented out?

**Review by vishnugopal — APPROVED** [review]
The squarespace import seems to work well for me. This also seems to make changes in Wix and Medium, but I didn't really test those because the changes seem to be small!

**Comment by zaerl** [comment]
> The Squarespace import seems to work well for me. This also seems to make changes in Wix and Medium, but I didn't really test those because the changes seem to be small!

Those are small changes. I've added `CompleteScreen` and `ProgressScreen` components which are shared between all routes. In order to avoid code duplication.

File: client/signup/steps/import-from/components/importer-drag/index.tsx (hunk 2 above)
Comment (zaerl) [reply client/signup/steps/import-from/components/importer-drag/index.tsx:49]: It is not needed now. Maybe shortly soon when we make a final optimization.

**Pull request merged**